pkg/authorizer/authorizer.go

```diff
@@ -0,0 +1,144 @@
+package authorizer
+
+import (
+	"fmt"
+	"net/http"
+	"net/url"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/3scale/3scale-go-client/threescale/api"
+	"github.com/3scale/3scale-istio-adapter/pkg/threescale/authorizer"
+	"github.com/3scale/3scale-porta-go-client/client"
+	log "github.com/sirupsen/logrus"
+)
+
+type Authorizer struct {
+	stopChan chan struct{}
+	config   Config
+	manager  *authorizer.Manager
+}
+
+type Config struct {
+	serviceAPICacheConfig *authorizer.SystemCacheConfig
+	authorizationAPICache bool
+	adminPortalURL        *url.URL
+	environment           string
+	serviceID             string
+}
+
+// TODO: Add a default builder that uses cache by default with everything set to sane defaults.
+// TODO: Add a more customizable builder.
+
+func New(AdminPortalURL *url.URL, environment, serviceID string, ServiceAPICache,
```

Comment:
What do you think about passing in configs here instead of bools. I see we set the config down later in the functionalists but doing that w have no way to external tweak it?

Reply:
Even further to that perhaps we can group the system info in a struct and pass it in, like env, url and config info, wdyt?